devito/ir/equations/algorithms.py

```diff
@@ -99,18 +102,30 @@ def lower_exprs(expressions, subs=None, **kwargs):
     --------
     f(x - 2*h_x, y) -> f[xi + 2, yi + 4]  (assuming halo_size=4)
     """
-    return _lower_exprs(expressions, subs or {})
+    return _lower_exprs(expressions, subs or {}, **kwargs)
 
 
-def _lower_exprs(expressions, subs):
+def _lower_exprs(expressions, subs, **kwargs):
+    # FIXME: Not sure why you can get this, but not access with index
+    sregistry = kwargs.get('sregistry')
+
     processed = []
+    rebuilt = {}
     for expr in as_tuple(expressions):
         try:
             dimension_map = expr.subdomain.dimension_map
         except AttributeError:
             # Some Relationals may be pure SymPy objects, thus lacking the subdomain
             dimension_map = {}
 
+        if sregistry and dimension_map:
+            # Give SubDimension thicknesses, SubDimensionSet functions and
+            # SubDomainSet implicit dimensions unique names
+            rename_thicknesses(dimension_map, sregistry, rebuilt)
+            # Rebuild ConditionalDimensions using rebuilt subdimensions
+            # The expression is then rebuilt with this ConditionalDimension
+            expr = rebuild_cdims(expr, rebuilt)
```

Comment:
imho `rename_thicknesses` and `rebuild_cdims` should all be part of the same dispatch logic; you may use a ranking system (see https://github.com/devitocodes/devito/blob/master/devito/passes/iet/engine.py#L412-L415 for an example) to ensure certain objects are rebuilt before others.

nitpicking: `uniquify_symbols` sounds like a better name to me